src/args_parser.c

```diff
@@ -2,10 +2,22 @@
 
 #include <stdlib.h>
 
-args_t args_parse(char **argv) {
+args_t args_parse(int argc, char **argv) {
     args_t parsed;
+    parsed.flags[0].flag_name = "";
+    parsed.flags[0].flag_value = "";
+    parsed.flags[1].flag_name = "";
+    parsed.flags[1].flag_value = "";
 
-    
+
+    for(int i = 0; i < 4 && i < argc-1; i = i + 2)
+    {
+        parsed.flags[i].flag_name = argv[i];
+        if(i+1 < argc-1)
+        {
+            parsed.flags[i].flag_value = argv[i+1];
```

Comment:
This is dummy verification as the user can use the program as:

```
./njvm --help --inform file.class
```

On the case presented the `--help` flag does not have a value so the `--inform` would be the the `--help` flag's value. This is wrong. A good verification should verify if the argument is a possible flag contained in the set of possible flags and parse its value after if such flag requires a value. Also, there's no way to know how many flags the user will set when calling the software.


This implementation might work on a certain level of usage but I have shown such use case where this wouldn't.